lib/_tls_wrap.js

```diff
@@ -374,6 +375,12 @@ TLSSocket.prototype._wrapHandle = function(wrap) {
     res = null;
   });
 
+  if (wrap instanceof Stream) {
```

Comment:
Hmm… does this play nicely with `readable-stream`? Or, equivalently, does the `socket instanceof Duplex` check in `TLSSocket()`?